src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -30,6 +34,8 @@ public static int Main(string[] args)
         public static int MainInternal(string[] args, CommandOutputLogger log)
         {
 #if DEBUG
+            MSBuildLocator.RegisterDefaults();
```

Comment:
Just learning about this locator, but one thing it said was to ensure this is called before any other MSBuild commands. Should it be the first line of Main just in case someone adds MSBuild code later and isn't aware? Also a comment mentioning this might be nice.

Reply:
It's already the first statement in `MainInternal`. I don't understand why we have a `MainInternal` method, but the only code that runs before is creating a new logger instance. It's even before the code to allow a debugger to attach for when you want to debug the `dotnet` cli integration. It would take a major redesign of the app for MSBuild to be called earlier, particularly as not all NuGet.CommandLine.XPlat commands use MSBuild.

So, I think the risk is very low, and the consequence would be that MSBuild would fail on SDK style projects, meaning it would be very obvious to the developer that their change doesn't work well.